verification/types.go

```diff
@@ -1,13 +1,42 @@
 package verification
 
-import "fmt"
+import (
+	"fmt"
+	nsigner "github.com/notaryproject/notation-core-go/signer"
+)
 
 // VerificationType is an enum for signature verification types such as Integrity, Authenticity, etc.
 type VerificationType string
 
 // VerificationAction is an enum for signature verification actions such as Enforced, Logged, Skipped.
 type VerificationAction string
 
+// VerificationResult encapsulates the verification result (passed or failed) for a verification type, including the
+// desired verification action as specified in the trust policy
+type VerificationResult struct {
+	// FailedToVerify is set to true if the verification is inconclusive due to a run time error
+	FailedToVerify bool
```

Comment:
Can we capture this condition as an error, this seems like an internal error condition in notation itself. You can set success to false, and capture details in error.

Reply:
If we do that, we won't be able to differentiate a bad signature from a good signature with a runtime error. Wouldn't the error message that is shown to the CLI users depend on this differentiation?

Reply:
Discussed offline. Removed `FailedToVerify`